### ISSUE TITLE
Test results are not collected

### ISSUE THREAD

**Issue by emartynov** [openbakery/gradle-xcodePlugin#202]
I have strange issue. Maybe it is misconfiguration or something.

With my test project https://github.com/emartynov/ios-app-with-gradle

I see in console 2 tests run, I see them also in xcode output, but xml with tests result is reporting about 0 test.

What is wrong?

**Comment by emartynov** [comment]
Adding part of my xcodebuild-output:
```
Test Suite 'All tests' started at 2015-08-27 10:25:36 +0000
Test Suite 'AppGradleTests.xctest' started at 2015-08-27 10:25:36 +0000
Test Suite 'AppGradleTests' started at 2015-08-27 10:25:36 +0000
Test Case '-[AppGradleTests testExample]' started.
Test Case '-[AppGradleTests testExample]' passed (0.001 seconds).
Test Case '-[AppGradleTests testPerformanceExample]' started.
/Users/310198568/Development/source/Own/AppGradle/AppGradleTests/AppGradleTests.m:37: Test Case '-[AppGradleTests testPerformanceExample]' measured [Time, seconds] average: 0.000, relative standard deviation: 114.170%, values: [0.000003, 0.000001, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000], performanceMetricID:com.apple.XCTPerformanceMetric_WallClockTime, baselineName: "", baselineAverage: , maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.100, maxStandardDeviation: 0.100
Test Case '-[AppGradleTests testPerformanceExample]' passed (0.303 seconds).
Test Suite 'AppGradleTests' passed at 2015-08-27 10:25:36 +0000.
	 Executed 2 tests, with 0 failures (0 unexpected) in 0.304 (0.305) seconds
Test Suite 'AppGradleTests.xctest' passed at 2015-08-27 10:25:36 +0000.
	 Executed 2 tests, with 0 failures (0 unexpected) in 0.304 (0.305) seconds
Test Suite 'All tests' passed at 2015-08-27 10:25:36 +0000.
	 Executed 2 tests, with 0 failures (0 unexpected) in 0.304 (0.306) seconds
[1m** TEST SUCCEEDED **

[0m
```
I went quickly through the code and can not find why it doesn't generate tests. I will try to run groovy about parsing output on this file

**Comment by emartynov** [comment]
Ah wait. Looks like this line makes something clear:
```
  <testsuite name='iPhone 4s/iOS Simulator/8.4' tests='0' errors='0' failures='0' skipped='0' />
```

**Comment by emartynov** [comment]
Ok,  more updates. If I remove destination than I see empty test cases for first sim:
```
<testsuites>
  <testsuite name='iPhone 4s/iOS Simulator/8.4' tests='0' errors='0' failures='0' skipped='0' />
  <testsuite name='iPhone 5/iOS Simulator/8.4' tests='2' errors='0' failures='0' skipped='0'>
    <testcase classname='AppGradleTests' name='testExample' time='0.0'>
      <system-out>Test Case '-[AppGradleTests testExample]' started.</system-out>
    </testcase>
    <testcase classname='AppGradleTests' name='testPerformanceExample' time='0.295'>
      <system-out>Test Case '-[AppGradleTests testPerformanceExample]' started.
/Users/310198568/Development/source/Own/AppGradle/AppGradleTests/AppGradleTests.m:37: Test Case '-[AppGradleTests testPerformanceExample]' measured [Time, seconds] average: 0.000, relative standard deviation: 90.967%, values: [0.000002, 0.000001, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000], performanceMetricID:com.apple.XCTPerformanceMetric_WallClockTime, baselineName: "", baselineAverage: , maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.100, maxStandardDeviation: 0.100</system-out>
    </testcase>
  </testsuite>
  <testsuite name='iPhone 5s/iOS Simulator/8.4' tests='2' errors='0' failures='0' skipped='0'>
    <testcase classname='AppGradleTests' name='testExample' time='0.0'>
      <system-out>Test Case '-[AppGradleTests testExample]' started.</system-out>
    </testcase>
    <testcase classname='AppGradleTests' name='testPerformanceExample' time='0.289'>
      <system-out>Test Case '-[AppGradleTests testPerformanceExample]' started.
/Users/310198568/Development/source/Own/AppGradle/AppGradleTests/AppGradleTests.m:37: Test Case '-[AppGradleTests testPerformanceExample]' measured [Time, seconds] average: 0.000, relative standard deviation: 98.710%, values: [0.000002, 0.000001, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000], performanceMetricID:com.apple.XCTPerformanceMetric_WallClockTime, baselineName: "", baselineAverage: , maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.100, maxStandardDeviation: 0.100</system-out>
    </testcase>
  </testsuite>
  <testsuite name='iPhone 6 Plus/iOS Simulator/8.4' tests='2' errors='0' failures='0' skipped='0'>
    <testcase classname='AppGradleTests' name='testExample' time='0.001'>
      <system-out>Test Case '-[AppGradleTests testExample]' started.</system-out>
    </testcase>
    <testcase classname='AppGradleTests' name='testPerformanceExample' time='0.31'>
      <system-out>Test Case '-[AppGradleTests testPerformanceExample]' started.
/Users/310198568/Development/source/Own/AppGradle/AppGradleTests/AppGradleTests.m:37: Test Case '-[AppGradleTests testPerformanceExample]' measured [Time, seconds] average: 0.000, relative standard deviation: 190.507%, values: [0.000005, 0.000001, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000], performanceMetricID:com.apple.XCTPerformanceMetric_WallClockTime, baselineName: "", baselineAverage: , maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.100, maxStandardDeviation: 0.100</system-out>
    </testcase>
  </testsuite>
  <testsuite name='iPhone 6/iOS Simulator/8.4' tests='2' errors='0' failures='0' skipped='0'>
    <testcase classname='AppGradleTests' name='testExample' time='0.0'>
      <system-out>Test Case '-[AppGradleTests testExample]' started.</system-out>
    </testcase>
    <testcase classname='AppGradleTests' name='testPerformanceExample' time='0.297'>
      <system-out>Test Case '-[AppGradleTests testPerformanceExample]' started.
/Users/310198568/Development/source/Own/AppGradle/AppGradleTests/AppGradleTests.m:37: Test Case '-[AppGradleTests testPerformanceExample]' measured [Time, seconds] average: 0.000, relative standard deviation: 119.900%, values: [0.000002, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000], performanceMetricID:com.apple.XCTPerformanceMetric_WallClockTime, baselineName: "", baselineAverage: , maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.100, maxStandardDeviation: 0.100</system-out>
    </testcase>
  </testsuite>
  <testsuite name='iPad 2/iOS Simulator/8.4' tests='2' errors='0' failures='0' skipped='0'>
    <testcase classname='AppGradleTests' name='testExample' time='0.0'>
      <system-out>Test Case '-[AppGradleTests testExample]' started.</system-out>
    </testcase>
    <testcase classname='AppGradleTests' name='testPerformanceExample' time='0.288'>
      <system-out>Test Case '-[AppGradleTests testPerformanceExample]' started.
/Users/310198568/Development/source/Own/AppGradle/AppGradleTests/AppGradleTests.m:37: Test Case '-[AppGradleTests testPerformanceExample]' measured [Time, seconds] average: 0.000, relative standard deviation: 92.249%, values: [0.000002, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000], performanceMetricID:com.apple.XCTPerformanceMetric_WallClockTime, baselineName: "", baselineAverage: , maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.100, maxStandardDeviation: 0.100</system-out>
    </testcase>
  </testsuite>
  <testsuite name='iPad Retina/iOS Simulator/8.4' tests='2' errors='0' failures='0' skipped='0'>
    <testcase classname='AppGradleTests' name='testExample' time='0.0'>
      <system-out>Test Case '-[AppGradleTests testExample]' started.</system-out>
    </testcase>
    <testcase classname='AppGradleTests' name='testPerformanceExample' time='0.291'>
      <system-out>Test Case '-[AppGradleTests testPerformanceExample]' started.
/Users/310198568/Development/source/Own/AppGradle/AppGradleTests/AppGradleTests.m:37: Test Case '-[AppGradleTests testPerformanceExample]' measured [Time, seconds] average: 0.000, relative standard deviation: 102.628%, values: [0.000002, 0.000001, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000], performanceMetricID:com.apple.XCTPerformanceMetric_WallClockTime, baselineName: "", baselineAverage: , maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.100, maxStandardDeviation: 0.100</system-out>
    </testcase>
  </testsuite>
  <testsuite name='iPad Air/iOS Simulator/8.4' tests='2' errors='0' failures='0' skipped='0'>
    <testcase classname='AppGradleTests' name='testExample' time='0.0'>
      <system-out>Test Case '-[AppGradleTests testExample]' started.</system-out>
    </testcase>
    <testcase classname='AppGradleTests' name='testPerformanceExample' time='0.307'>
      <system-out>Test Case '-[AppGradleTests testPerformanceExample]' started.
/Users/310198568/Development/source/Own/AppGradle/AppGradleTests/AppGradleTests.m:37: Test Case '-[AppGradleTests testPerformanceExample]' measured [Time, seconds] average: 0.000, relative standard deviation: 107.740%, values: [0.000002, 0.000001, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000], performanceMetricID:com.apple.XCTPerformanceMetric_WallClockTime, baselineName: "", baselineAverage: , maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.100, maxStandardDeviation: 0.100</system-out>
    </testcase>
  </testsuite>
  <testsuite name='Resizable iPhone/iOS Simulator/8.4' tests='2' errors='0' failures='0' skipped='0'>
    <testcase classname='AppGradleTests' name='testExample' time='0.001'>
      <system-out>Test Case '-[AppGradleTests testExample]' started.</system-out>
    </testcase>
    <testcase classname='AppGradleTests' name='testPerformanceExample' time='0.308'>
      <system-out>Test Case '-[AppGradleTests testPerformanceExample]' started.
/Users/310198568/Development/source/Own/AppGradle/AppGradleTests/AppGradleTests.m:37: Test Case '-[AppGradleTests testPerformanceExample]' measured [Time, seconds] average: 0.000, relative standard deviation: 102.563%, values: [0.000003, 0.000001, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000], performanceMetricID:com.apple.XCTPerformanceMetric_WallClockTime, baselineName: "", baselineAverage: , maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.100, maxStandardDeviation: 0.100</system-out>
    </testcase>
  </testsuite>
  <testsuite name='Resizable iPad/iOS Simulator/8.4' tests='2' errors='0' failures='0' skipped='0'>
    <testcase classname='AppGradleTests' name='testExample' time='0.0'>
      <system-out>Test Case '-[AppGradleTests testExample]' started.</system-out>
    </testcase>
    <testcase classname='AppGradleTests' name='testPerformanceExample' time='0.303'>
      <system-out>Test Case '-[AppGradleTests testPerformanceExample]' started.
/Users/310198568/Development/source/Own/AppGradle/AppGradleTests/AppGradleTests.m:37: Test Case '-[AppGradleTests testPerformanceExample]' measured [Time, seconds] average: 0.000, relative standard deviation: 143.509%, values: [0.000003, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000], performanceMetricID:com.apple.XCTPerformanceMetric_WallClockTime, baselineName: "", baselineAverage: , maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.100, maxStandardDeviation: 0.100</system-out>
    </testcase>
  </testsuite>
</testsuites>
```

I was thinking this is specific for iPhone4 sim. But looks like it is specific for number of sims to run and first one is always empty:
```
<testsuites>
  <testsuite name='iPhone 5s/iOS Simulator/8.4' tests='0' errors='0' failures='0' skipped='0' />
  <testsuite name='iPhone 4s/iOS Simulator/8.4' tests='2' errors='0' failures='0' skipped='0'>
    <testcase classname='AppGradleTests' name='testExample' time='0.001'>
      <system-out>Test Case '-[AppGradleTests testExample]' started.</system-out>
    </testcase>
    <testcase classname='AppGradleTests' name='testPerformanceExample' time='0.317'>
      <system-out>Test Case '-[AppGradleTests testPerformanceExample]' started.
/Users/310198568/Development/source/Own/AppGradle/AppGradleTests/AppGradleTests.m:37: Test Case '-[AppGradleTests testPerformanceExample]' measured [Time, seconds] average: 0.000, relative standard deviation: 119.711%, values: [0.000002, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000, 0.000000], performanceMetricID:com.apple.XCTPerformanceMetric_WallClockTime, baselineName: "", baselineAverage: , maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.100, maxStandardDeviation: 0.100</system-out>
    </testcase>
  </testsuite>
</testsuites>
```
I hope it gives some insights. I will step to the code of plugin later

**Comment by renep** [comment]
I have tested it with my version and it looks fine there, but with the current 0.11.4 I can reproduce your issue.

I have created a test version with some changes: xcode-plugin:0.11.5-TEST

<code>
	dependencies {
	    classpath "org.openbakery:xcode-plugin:0.11.5-TEST"
	}
</code>

Can you please test this version.

**Comment by emartynov** [comment]
Yes!

Works for me. Thank you and waiting for update!

**Comment by renep** [comment]
fixed with 0.11.5
